lib/mayaUsd/nodes/layerManager.cpp

```diff
@@ -0,0 +1,909 @@
+            continue;
+        }
+
+        bool layerContainsEdits = true;
+        serializedVal = serializedPlug.asString(MDGContext::fsNormal, &status).asChar();
+        if (serializedVal.empty()) {
+            layerContainsEdits = false;
+        }
+
+        bool isAnon = anonymousPlug.asBool(MDGContext::fsNormal, &status);
+        if (isAnon) {
+            // Note that the new identifier will not match the old identifier - only the "tag"
+            // will be retained
+            layer
+                = SdfLayer::CreateAnonymous(SdfLayer::GetDisplayNameFromIdentifier(identifierVal));
+        } else {
+            SdfLayerHandle layerHandle = SdfLayer::Find(identifierVal);
+            if (layerHandle) {
+                layer = layerHandle;
+            } else {
+                // TODO: currently, there is a small window here, after the find, and before the
+                // New, where another process might sneak in and create a layer with the same
+                // identifier, which could cause an error. This seems unlikely, but we have a
+                // discussion with Pixar to find a way to avoid this.
+
+                SdfFileFormatConstPtr fileFormat
+                    = getFileFormatForLayer(identifierVal, serializedVal);
+
+                if (layerContainsEdits) {
+                    // In order to make the layer reloadable by SdfLayer::Reload(), we hack the
+                    // identifier with temp one on creation and call layer->SetIdentifier()
+                    // again to set the timestamp:
+                    layer = SdfLayer::New(fileFormat, identifierVal + identifierTempSuffix);
+                    if (!layer) {
+                        MGlobal::displayError(
+                            MString("Error - failed to create new layer for identifier '")
+                            + identifierVal.c_str() + "' for plug " + idPlug.partialName(true));
+                        continue;
+                    }
+                    layer->SetIdentifier(
+                        identifierVal); // Make it reloadable by SdfLayer::Reload(true)
+                    layer->Clear();     // Mark it dirty to make it reloadable by SdfLayer::Reload()
+                                        // without force=true
+                } else {
+                    layer = SdfLayer::FindOrOpen(identifierVal);
+                }
+            }
+        }
+
+        if (layer) {
+            if (layerContainsEdits) {
+                if (!layer->ImportFromString(serializedVal)) {
+                    MGlobal::displayError(
+                        MString("Failed to import serialized layer: ") + serializedVal.c_str());
+                    continue;
+                }
+            }
+
+            LayerDatabase::instance().addLayer(layer, identifierVal);
+            createdLayers.push_back(layer);
+        }
+    }
+
+    removeManagerNode(lm);
```

Comment:
It would be great to mark the maya scene as clean once the layer manager node is removed so that the "unsaved scene" dialog doesn't pop up after opening a scene that contains a layer manager node and opening a new scene (or closing maya) right after.

Reply:
Good point, I'll look into that.  Can't remember off the top of my head if there's a MGlobal/MFileIO call to easily do that or not...